libglusterfs/src/inode.c

```diff
@@ -137,6 +137,21 @@ inode_table_prune(inode_table_t *table);
 void
 fd_dump(struct list_head *head, char *prefix);
 
+/* Calculate index based on the inode_table root_level
+   and xlator_id
+*/
+static int
+inode_get_ctx_index(inode_t *inode, xlator_t *xlator)
+{
+    int ctx_idx = xlator->level;
+
+    if (ctx_idx > inode->table->root_level)
+        ctx_idx = (inode->table->root_level + xlator->xl_id -
+                   inode->table->root_id);
```

Comment:
I think it would be a good idea to add a check for debug builds like:

```c
#ifdef DEBUG
    int32_t idx = xlator->xl_id - inode->table->root_id;
    GF_ASSERT((idx > 0) && (idx < (inode->table->ctxcount - inode->table->root_level)));
#endif
```

Reply:
done